### PR TITLE
feat(protocol): add note about partial updates

### PR DESCRIPTION
**Pull request by coderbyheart**
This adds a note about the devices being allowed (and encourage) to send partial updates.

### CODE REVIEW

File: docs/cloud-protocol/DeviceDataAndConfiguration.rst [hunk 1]
```diff
@@ -7,6 +7,13 @@ Device data and configuration
    :local:
    :depth: 2
 
+The schemas referenced below define the data format used by the device to encode data.
+Currently :ref:`JSON is the only supported encoding format <json-format>`.
```
Comment (peknis) [inline docs/cloud-protocol/DeviceDataAndConfiguration.rst]:
```suggestion
Currently, :ref:`JSON <json-format>` is the only supported encoding format.
```
Just a proposal. Works as it is, but I'd rather have it like this.

File: docs/cloud-protocol/DeviceDataAndConfiguration.rst [hunk 2]
```diff
@@ -7,6 +7,13 @@ Device data and configuration
...
+
+While the schemas mark properties as required, the device can always send partial updates.
+For example, the device might initially after boot report the full roaming information, but later only send the changed values (typically RSRP) to the cloud in order conserve data and power.
```
Comment (peknis) [inline docs/cloud-protocol/DeviceDataAndConfiguration.rst]:
```suggestion
For example, the device might initially report the full roaming information after boot, but later only send the changed values (typically RSRP) to the cloud to conserve data and power.
```
Should we open the abbreviation?

File: docs/cloud-protocol/DeviceDataAndConfiguration.rst [hunk 3]
```diff
@@ -7,6 +7,13 @@ Device data and configuration
...
+It is the responsibility of the cloud to store the updates in a way that full objects can be constructed from partial updates, when needed.
```
Comment (peknis) [inline docs/cloud-protocol/DeviceDataAndConfiguration.rst]: Do we need to say "when needed"?

File: docs/cloud-protocol/DeviceDataAndConfiguration.rst [hunk 4]
```diff
@@ -7,6 +7,13 @@ Device data and configuration
    :local:
    :depth: 2
 
+The schemas referenced below define the data format used by the device to encode data.
```
Comment (peknis) [inline docs/cloud-protocol/DeviceDataAndConfiguration.rst]:
```suggestion
The schemas referenced on this page define the data format used by the device to encode data.
```

**Review by peknis — CHANGES_REQUESTED** [review]

File: docs/cloud-protocol/DeviceDataAndConfiguration.rst (hunk 3 above)
Comment (coderbyheart) [reply docs/cloud-protocol/DeviceDataAndConfiguration.rst]: Not really, it's implied.

Comment (coderbyheart) [inline docs/cloud-protocol/DeviceDataAndConfiguration.rst]:
```suggestion
It is the responsibility of the cloud to store the updates in a way that full objects can be constructed from partial updates.
```

File: docs/cloud-protocol/DeviceDataAndConfiguration.rst [hunk 5]
```diff
@@ -7,6 +7,13 @@ Device data and configuration
    :local:
    :depth: 2
 
+The schemas referenced on this page define the data format used by the device to encode data.
+Currently, :ref:`JSON <json-format>` is the only supported encoding format.
+
+While the schemas mark properties as required, the device can always send partial updates.
+For example, the device might initially report the full roaming information after boot, but later only send the changed values (typically RSRP) to the cloud to conserve data and power.
```
Comment (peknis) [inline docs/cloud-protocol/DeviceDataAndConfiguration.rst]: What about the abbreviation RSRP? Would it be worth opening?

Comment (coderbyheart) [inline docs/cloud-protocol/DeviceDataAndConfiguration.rst]:
```suggestion
For example, the device might initially report the full roaming information after boot, but later only send the changed values, for example Reference Signal Received Power (RSRP), to the cloud to conserve data and power.
```

Comment (coderbyheart) [reply docs/cloud-protocol/DeviceDataAndConfiguration.rst]: It's explained in the schema files, but good idea to explain it here.

**Review by wiba-nordic — APPROVED** [review]

**Review by peknis — APPROVED** [review]

**Review by simensrostad — APPROVED** [review]

**Pull request merged**